### PR TITLE
refactor(api): tools: handle args better and inject driver and hw

### PR DESCRIPTION
**Pull request by sfoster1**
The way the tools work, with building a driver in the module __init__,
leads to weirdness around argument parsing and requires import-time
initialization. Fix that as much as possible.

## Risk Assessment

Definitely run through the factory testing, but this shouldn't affect anything else.

### CODE REVIEW

File: api/src/opentrons/tools/args_handler.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from typing import Tuple
+
+from opentrons.hardware_control import ThreadManager, API, adapters
+from opentrons.drivers.smoothie_drivers.driver_3_0 import SmoothieDriver_3_0_0
+
+
+def root_argparser():
+    parse = argparse.ArgumentParser()
+    parse.add_argument('-p', '--port',
+                       description='serial port of the smoothie',
```
Comment (Laura-Danielle) [inline api/src/opentrons/tools/args_handler.py]: It's dumb but the argument for description is actually called `help`. See [here](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument)

File: api/src/opentrons/tools/args_handler.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                       default='', type='str')
```
Comment (Laura-Danielle) [inline api/src/opentrons/tools/args_handler.py]: type needs to be a callable.

File: api/src/opentrons/tools/z_stage_test.py [hunk 3]
```diff
@@ -80,89 +79,84 @@ def test_axis(axis, tolerance):
...
         dest="low_current",
         type="float",
         default=0.05,
         help="low z motor current")
-    parser.add_option(
+    parser.add_argument(
         "-c",
         "--cycles",
         dest="cycles",
         type="int",
```
Comment (Laura-Danielle) [inline api/src/opentrons/tools/z_stage_test.py]: type needs to be callable

File: api/src/opentrons/tools/z_stage_test.py [hunk 4]
```diff
@@ -80,89 +79,84 @@ def test_axis(axis, tolerance):
...
         default=2,
         help="Cycles to run test")
-    parser.add_option(
+    parser.add_argument(
         "-t",
         "--tolerance",
         dest="tolerance",
         type="float",
```
Comment (Laura-Danielle) [inline api/src/opentrons/tools/z_stage_test.py]: Needs to be callable

File: api/src/opentrons/tools/z_stage_test.py [hunk 5]
```diff
@@ -80,89 +79,84 @@ def test_axis(axis, tolerance):
...
         try:
             print("Testing A")
-            test_axis('A', options.tolerance)
+            test_axis('A', options.tolerance, driver)
         except Exception as e:
             print("FAIL: {}".format(e))
 
 
 if __name__ == '__main__':
     atexit.register(_exit_test)
```
Comment (Laura-Danielle) [inline api/src/opentrons/tools/z_stage_test.py]: Gotta pass in driver as an argument.

**Comment by sfoster1** [comment]
> After I made the changes I specified, everything worked except the camera image.
> 
> When I tried to access the link specified: `192.168.1.12:8000/cam_test.jpg` it said connection refused.

Forgot to actually push the changes for that, whoops. Weirdly the cam test thing worked for me

File: api/src/opentrons/tools/z_stage_test.py [hunk 6]
```diff
@@ -80,89 +80,85 @@ def test_axis(axis, tolerance):
...
 
 
 if __name__ == '__main__':
-    atexit.register(_exit_test)
-    options, args = get_options()
+    parser = args_handler.root_argparser(
+        "Test the z-stages of the OT-2")
+    args = get_options(parser)
+    _, driver = args_handler.build_driver(args.port)
+    atexit.register(functools.partial(_exit_test, driver))
```
Comment (Laura-Danielle) [inline api/src/opentrons/tools/z_stage_test.py]: Not important to change, but you can actually pass in the arguments for _exit_test as an argument for the r[egister method](https://docs.python.org/3/library/atexit.html)

Comment (sfoster1) [reply api/src/opentrons/tools/z_stage_test.py]: good catch, probably will keep it a partial since there's a lot going on tho

**Pull request merged**